services/sharepoint-connector/deploy/Dockerfile

```diff
@@ -0,0 +1,67 @@
+# syntax=docker/dockerfile:1.7
+
+FROM node:22-bookworm AS base
+ENV PNPM_HOME="/pnpm"
+ENV PATH="$PNPM_HOME:$PATH"
+RUN corepack enable
+
+FROM base AS pruner
+WORKDIR /app
+ARG APP_NAME
+RUN --mount=type=bind,source=.,target=/src,ro \
+    pnpm dlx turbo prune @unique-ag/${APP_NAME} --docker --cwd /src --out-dir /app/out
+
+FROM base AS deps
+WORKDIR /app
+
+COPY --from=pruner /app/out/json/ .
+COPY ./pnpm-lock.yaml ./pnpm-lock.yaml
+COPY ./pnpm-workspace.yaml ./pnpm-workspace.yaml
+
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store \
+    pnpm install --frozen-lockfile
+
+FROM base AS builder
+ARG APP_NAME
+WORKDIR /app
+
+COPY --from=deps /app/ .
+COPY ./tsconfig.json ./tsconfig.json
+COPY ./turbo.json ./turbo.json
+COPY --from=pruner /app/out/full/ .
+
+RUN pnpm --filter=@unique-ag/${APP_NAME} codegen
+RUN --mount=type=cache,target=/root/.cache/turbo \
+    pnpm turbo build --filter=@unique-ag/${APP_NAME}... && \
+    pnpm --filter=@unique-ag/${APP_NAME} deploy out
+
+FROM node:22-bookworm-slim AS runner
```

Comment:
here as well, the existing dockerfile pins the pnpm version: ARG PNPM_VERSION=10.15.1